### DISCUSSION TITLE
Preferences Button on Mac shows Cmd + , instead of Cmd + U

### DISCUSSION THREAD

**Discussion by MsBlockCat** [opentoonz/opentoonz#6546] [General]
### Issue Summary
OpenToonz > Preferences on Macs shows Cmd + , as a keyboard shortcut, but since this keyboard shortcut is already assigned to something else it doesn't open Preferences. Pressing Cmd + U is what actually opens Preferences.

### Screenshot or Video Reference
<img width="268" alt="Screen Shot 2020-03-02 at 10 24 31 PM" src="https://user-images.githubusercontent.com/37186692/75740032-98869380-5cd4-11ea-8c40-5d0b499d2ca3.png">

### Expected Results
The keyboard shortcut should say it's Cmd + U

### Actual Results
The keyboard shortcut says it's Cmd + ,

### System Information
* **OpenToonz Version**: 1.4.0
* **Operating System**: macOS Catalina
* **CPU**: Quad Core i5
* **Memory**: 8 GB LPDDR3
* **Graphics Card**: 
* **Graphics Tablet**:

**Comment by artisteacher** [comment]
I think this might have to do with the QT macOS menu roles. About, Preferences and Quit are supposed to be set to QAction::AboutRole, QAction::PreferencesRole, and QAction::QuitRole. That automatically places the menu items in the expected part of the application menu. Other menu items usually are set to QAction::NoRole to prevent them from ending up in the Applications menu (due to mac naming conventions). This is set in mainwindow.cpp  (toonz/sources/toonz/mainwindow.cpp):
<img width="461" alt="Screen Shot 2020-03-07 at 9 34 47 PM" src="https://user-images.githubusercontent.com/21159116/76155554-7ff2f080-60bb-11ea-97b9-9202a2ffb42f.png">

The shortcut is set properly in mainwindow.cpp:
<img width="618" alt="Screen Shot 2020-03-07 at 9 32 11 PM" src="https://user-images.githubusercontent.com/21159116/76155544-4c17cb00-60bb-11ea-9ada-d6545772900c.png">

`Cmd+,` is the expected shortcut to open Preferences on applications in macOS. Using QAction::PreferencesRole must automatically add that to the display of the menu item, even though the actual shortcut does not change from Cmd+U...

Other possible roles to place something in the application menu are "QAction::TextHeuristicRole" (which seems to behave no differently from the preference role in this case) and "QAction::ApplicationSpecificRole". Using the latter, the shortcut is correct but the preferences menu item displays 8 times:
<img width="186" alt="Screen Shot 2020-03-07 at 9 50 39 PM" src="https://user-images.githubusercontent.com/21159116/76155709-c9443f80-60bd-11ea-98dc-5b0c17b6f54d.png">
I'm wondering if this has to do with preferences being listed 7 times in menubar.cpp?

**Comment by MsBlockCat** [comment]
Interesting! This code mostly goes well above my head, but if I had to guess Preferences being listed 7 times would probably cause it since it sounds like the 8th instance is there by default. Does anything seem to break if you do remove the extras in your test environment? I wonder why they were added...

**Comment by artisteacher** [comment]
Commenting out preferences from the file menu (line 1120) stops the preferences menu item from displaying at all. The other occurrences seem to have no impact. If only the file menu preference is active, it still displays multiple times.

**Comment by MsBlockCat** [comment]
@RodneyBaker I've added a $10 bounty to fix this issue! https://www.bountysource.com/issues/89666431-preferences-button-on-mac-shows-cmd-instead-of-cmd-u
